cranelift-codegen/meta/src/isa/x86/recipes.rs

```diff
@@ -2302,6 +2303,21 @@ pub fn define<'shared>(
             ),
     );
 
+    recipes.add_template_recipe(
+        EncodingRecipeBuilder::new("vconst", f_unary_imm128, 5)
+            .operands_out(vec![fpr])
+            .clobbers_flags(false)
+            .emit(
+                r#"
+                    {{PUT_OP}}(bits, rex2(0, out_reg0), sink);
+                    modrm_riprel(out_reg0, sink);
+                    // No reloc is needed here as the jump table is emitted directly after
+                    // the function body.
+                    const_disp4(imm, func, sink);
+                "#,
+            ),
```

Comment:
Here is the crux of the issue: when the recipe is being used to encode the instruction, it has my new `Uimm128` type (it is boxed, but that's a separate issue). However, this type is not a `Constant`, which must have an offset attached at some point. I can convert from `Uimm128` to `Constant` using `From` but obviously the offset will be `None` so when I inform the embedder that there is a `Constant` to relocate they will not have an offset to calculate with. I think I want to retain the `Uimm128` type because it seems to fit well with the format/operand kind world but `Function` and its related classes all probably need something more generic (i.e. `Constant`). What to do?